src/plugins/search/search.js

```diff
@@ -41,95 +158,135 @@ function Search(instance) {
+    this.callback = newCallback;
+  }
 
-  var defaultCallback = Search.DEFAULT_CALLBACK;
-  var defaultQueryMethod = Search.DEFAULT_QUERY_METHOD;
-  var defaultSearchResultClass = Search.DEFAULT_SEARCH_RESULT_CLASS;
+  /**
+   * Get queryMethod function.
+   *
+   * @returns {Function} Return the query method.
+   */
+  getQueryMethod() {
+    return this.queryMethod;
+  }
 
-  return {
-    getDefaultCallback() {
-      return defaultCallback;
-    },
+  /**
+   * Set queryMethod function.
+   *
+   * @param {Function} newQueryMethod
+   */
+  setQueryMethod(newQueryMethod) {
+    this.queryMethod = newQueryMethod;
+  }
 
-    setDefaultCallback(newDefaultCallback) {
-      defaultCallback = newDefaultCallback;
-    },
+  /**
+   * Get search result cells class.
+   *
+   * @returns {Function} Return the cell class.
+   */
+  getSearchResultClass() {
+    return this.searchResultClass;
+  }
 
-    getDefaultQueryMethod() {
-      return defaultQueryMethod;
-    },
+  /**
+   * Set search result cells class.
+   *
+   * @param {String} newElementClass
+   */
+  setSearchResultClass(newElementClass) {
+    this.searchResultClass = newElementClass;
+  }
 
-    setDefaultQueryMethod(newDefaultQueryMethod) {
-      defaultQueryMethod = newDefaultQueryMethod;
-    },
+  /**
+   * Checks the settings of the plugin.
+   *
+   * @param {Object} searchSettings The plugin settings, taken from Handsontable configuration.
+   * @private
+   */
+  checkPluginSettings(searchSettings) {
+    if (isObject(searchSettings)) {
+      if (searchSettings.searchResultClass) {
+        this.setSearchResultClass(searchSettings.searchResultClass);
+      }
 
-    getDefaultSearchResultClass() {
-      return defaultSearchResultClass;
-    },
+      if (searchSettings.queryMethod) {
+        this.setQueryMethod(searchSettings.queryMethod);
+      }
 
-    setDefaultSearchResultClass(newSearchResultClass) {
-      defaultSearchResultClass = newSearchResultClass;
+      if (searchSettings.callback) {
+        this.setCallback(searchSettings.callback);
+      }
     }
-  };
+  }
 
-}());
+  /** *
+   * The `beforeRenderer` hook callback.
+   *
+   * @private
+   * @param {HTMLTableCellElement} TD The rendered `TD` element.
+   * @param {Number} row Visual row index.
+   * @param {Number} col Visual column index.
+   * @param {String | Number} prop Column property name or a column index, if datasource is an array of arrays.
+   * @param {String} value Value of the rendered cell.
+   * @param {Object} cellProperties Object containing the cell's properties.
+   */
+  onBeforeRenderer(TD, row, col, prop, value, cellProperties) {
+    let classArray = [];
 
-function SearchCellDecorator(instance, TD, row, col, prop, value, cellProperties) {
-  var searchResultClass = (cellProperties.search !== null && typeof cellProperties.search == 'object' &&
-      cellProperties.search.searchResultClass) || Search.global.getDefaultSearchResultClass();
+    if (cellProperties.className) {
```

Comment:
Please add tests. Please cache the `cellProperties.className`.

Reply:
Done.